internal/blocksync/reactor.go

```diff
@@ -545,37 +553,35 @@ func (r *Reactor) poolRoutine(ctx context.Context, stateSynced bool, blockSyncCh
 						return
 					}
 				}
-			} else {
-				r.pool.PopRequest()
+				return
+			}
 
-				// TODO: batch saves so we do not persist to disk every block
-				r.store.SaveBlock(first, firstParts, second.LastCommit)
+			r.pool.PopRequest()
 
-				var err error
```

Comment:
Well we could still one day batch save blocks

Reply:
Yeah I think you would have to batch save before applying any of the blocks.

I don't think it's too much of an issue if you crash so long as the application height never gets ahead of the block store height

Reply:
> On the other hand, what about the "zombie TODO" below? I don't even understand what it means, do you?

I think it just means we should implement a more graceful way of dealing with a failure in `ApplyBlock` instead of just panicking